packages/web/src/components/buy-audio-modal/components/InProgressPage.tsx

```diff
@@ -0,0 +1,106 @@
+import {
+  buyAudioSelectors,
+  BuyAudioStage,
+  formatNumberString
+} from '@audius/common'
+import { IconCaretDown } from '@audius/stems'
+import { useSelector } from 'react-redux'
+
+import LoadingSpinner from 'components/loading-spinner/LoadingSpinner'
+import { ToggleCollapseButton } from 'components/toggle-collapse-button'
+
+import { IconAUDIO, IconSOL, IconUSD } from './Icons'
+import styles from './InProgressPage.module.css'
+
+const { getAudioPurchaseInfo, getBuyAudioFlowStage } = buyAudioSelectors
+
+const messages = {
+  pleaseHold: 'Please hold on. This may take a few moments.',
+  completeWithCoinbase: 'Complete this step with Coinbase',
+  step1: 'Step 1',
+  step2: 'Step 2',
+  purchasingSol: 'Purchasing SOL',
+  convertingSol: 'Converting SOL to $AUDIO',
+  moreInfo: 'More Info',
+  lessInfo: 'Less Info',
+  usd: 'USD',
+  sol: 'SOL',
+  audio: '$AUDIO'
+}
+
+export const InProgressPage = () => {
+  const purchaseInfo = useSelector(getAudioPurchaseInfo)
+  const buyAudioFlowStage = useSelector(getBuyAudioFlowStage)
+  const isStepOne = buyAudioFlowStage === BuyAudioStage.PURCHASING
+  let firstToken, secondToken
```

Comment:
type these potentially? also i think to do that it should be separate lets, which i also think is more standard